File: filelist/src/main/scala/farjs/filelist/fs/FSFileListActions.scala
package farjs.filelist.fs

import farjs.filelist.FileListActions
import farjs.filelist.api.FileListApi

import scala.concurrent.ExecutionContext.Implicits.global
import scala.concurrent.Future

class FSFileListActions private[fs](
  protected val api: FileListApi,
  fsService: FSService
) extends FileListActions {

  def getDriveRoot(path: String): Future[Option[String]] = {
    fsService.readDisk(path).map(_.map(_.root))
  }
}

object FSFileListActions extends FSFileListActions(new FSFileListApi, FSService.instance)

File: fs/FSFileListActions.mjs
/**
 * @import FileListApi from "@farjs/filelist/api/FileListApi.mjs"
 */
import FileListActions from "@farjs/filelist/FileListActions.mjs";
import FSFileListApi from "./FSFileListApi.mjs";

/**
 *
 * @param {FileListApi} api
 * @returns {FileListActions}
 */
function FSFileListActions(api = new FSFileListApi()) {
  return new FileListActions(api);
}

/** @type {FileListActions} */
FSFileListActions.instance = FSFileListActions();

export default FSFileListActions;
